src/main/typescript/components/login/LoginPage.tsx

```diff
@@ -17,38 +17,64 @@ import * as React from "react"
 import { Component } from "react"
 import { ReduxAction } from "../../lib/redux"
 import { connect } from "react-redux"
-import { authenticate } from "../../actions/authenticationActions"
+import { authenticate } from "../../actions/userActions"
 import { Redirect, RouteComponentProps } from "react-router"
 import { AppState } from "../../model/AppState"
 import { homeRoute } from "../../constants/clientRoutes"
+import { InjectedFormProps, reduxForm } from "redux-form"
 
 interface LoginPageProps {
-    authenticate: () => ReduxAction<Promise<void>>
+    authenticate: (username: string, password: string) => ReduxAction<Promise<any>>
     authenticated: boolean
+    authenticating: boolean
+    errorMessage: Error
 }
 
-class LoginPage extends Component<LoginPageProps & RouteComponentProps<any>> {
-    constructor(props: LoginPageProps & RouteComponentProps<any>) {
+interface LoginPageState {
+    loginName: string
+    loginPassword: string
+}
+
+type AllDemoFormProps = LoginPageProps & RouteComponentProps<any> & InjectedFormProps<LoginPageState>
+
+class LoginPage extends Component<AllDemoFormProps> {
+    constructor(props: AllDemoFormProps) {
         super(props)
+        this.state = {loginName: "", loginPassword: ""}
     }
 
-    login = () => this.props.authenticate()
+    showResults = (values: LoginPageState) => {
+        console.log(values);
+        this.props.authenticate(values.loginName, values.loginPassword)
+    }
 
     render() {
-        const { authenticated, location } = this.props
-        const { from } = location.state || { from: { pathname: homeRoute } }
+        const {authenticated, errorMessage, location, handleSubmit} = this.props
+        const {from} = location.state || {from: {pathname: homeRoute}}
 
         return authenticated
             ? <Redirect to={from}/>
-            : <div>
+            : <form onSubmit={handleSubmit(this.showResults)}>
                 <p>You must log in to view this page at {from.pathname}</p>
-                <button onClick={this.login}>Log in</button>
-            </div>
+                <label>Username</label>
+                <input type="text" name="loginName" required/>
+                <br/>
+                <label>Password</label>
+                <input type="password" name="loginPassword" required/>
+                <br/>
+                {this.props.error && <span>{this.props.error}<br/></span>}
```

Comment:
1. why not extract the `error` from the props above?
2. what's the difference between `error` here and `errorMessage` below?